jerry-core/jcontext/jcontext.h

```diff
@@ -35,6 +36,18 @@
 #define JERRY_CONTEXT_FIRST_MEMBER ecma_builtin_objects
 
 /**
+ * User context item
+ */
+typedef struct jerry_context_data_header
+{
+  struct jerry_context_data_header *next_p; /**< pointer to next context item */
+  const jerry_context_data_manager_t *manager_p; /**< manager responsible for deleting this item */
+} jerry_context_data_header_t;
+
+#define JERRY_CONTEXT_DATA_HEADER_USER_DATA(item_p) \
+  (&(((char *) item_p)[sizeof (jerry_context_data_header_t)]))
```

Comment:
I'd go with ((jerry_context_data_header_t *) item_p) to make sure that the "+1" has the right semantics.